examples/cloud_discovery_service/library_api/c++11/cloud_discovery_service_library_api.cxx

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ (c) 2005-2023 Copyright, Real-Time Innovations, Inc. All rights reserved.
+ RTI grants Licensee a license to use, modify, compile, and create derivative
+ works of the Software. Licensee has the right to distribute object form only
+ for use with RTI products. The Software is provided "as is", with no warranty
+ of any type, including any warranty for fitness for any purpose. RTI is under
+ no obligation to maintain or support the Software. RTI shall not be liable for
+ any incidental or consequential damages arising out of the use or inability to
+ use the software.
+ ******************************************************************************/
+
+#include <iostream>
+#include <thread>
+#include <chrono>
+#include <map>
+
+#include <rti/cds/Service.hpp>
+
+void service_main(std::string cfg_name, std::string cfg_file)
+{
+    rti::cds::Service service(
+            rti::cds::ServiceProperty()
+                    .cfg_file(cfg_file)
+                    .service_name(cfg_name)
+                    .application_name(
+                            "CloudDiscoveryService-LibraryAPI-CXX11"));
+
+    //  Dial up the logging at service level using the public logger API
+    rti::cds::Logger::instance().service_verbosity(
+            static_cast<rti::config::Verbosity::type>(
+                    rti::config::Verbosity::STATUS_ALL));
+
+    service.start();
+
+    std::cout << "ConfigName=" << cfg_name << ", "
+              << "ConfigFile=" << cfg_file << std::endl;
+    // Infinite loop until the user terminates the process
+    while (1) {
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
```

Comment:
We should have an exit condition  (handling CTRL+C) and maybe a timeout option, but the latter wouldn't be needed.

Reply:
I added a `runForSecs` parameter as we discussed on Slack.